### PR TITLE
DOC-2558: New comments with mentions page and demos

### PR DESCRIPTION
**Pull request by MitchC1999**
Ticket: DOC-2558

Site: [Staging branch](http://docs-feature-doc-2558.staging.tiny.cloud/docs/tinymce/latest/comments-with-mentions/)

Changes:
* Added Comments with mentions demos
* New Comments with mentions page
* New `tinycomments_mentions_enabled` option partial file

Pre-checks:
- [x] Branch prefixed with `feature/`, 
- [x] `modules/ROOT/nav.adoc` has been updated `(if applicable)`.

- [x] Remember to revert the version livedemo url

Review:
- [x] Documentation Team Lead has reviewed

### CODE REVIEW

File: modules/ROOT/partials/plugin-apis/comments-apis.adoc [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+const comments = tinymce.activeEditor.plugins.tinycomments;
+
+console.log(comments.getEventLog());
+console.log(comments.getEventLog(
+  { after: '2022-02-22T12:34:56Z' }  // ISO-8601 standard: YYYY-MM-DDThh:mm:ssZ
+));
+
+const eventLog = comments.getEventLog();
+const events = eventLog.events;
+const mentionedUsers = events.flatMap(({ mentionUids }) => mentionedUids);
```
Comment (kemister85) [inline modules/ROOT/partials/plugin-apis/comments-apis.adoc:134]:
```suggestion
const mentionedUsers = events.flatMap(({ mentionedUids }) => mentionedUids);
```

**Review by kemister85 — COMMENTED** [review]
Suggested copy edits.

File: antora.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ asciidoc:
     companyurl: https://www.tiny.cloud
     cdnurl: https://cdn.tiny.cloud/1/no-api-key/tinymce/7/tinymce.min.js
     tdcdnurl: https://cdn.tiny.cloud/1/_your_api_key_/tinydrive/7/tinydrive.min.js
-    tinymce_live_demo_url: https://cdn.tiny.cloud/1/qagffr3pkuv17a8on1afax661irst1hbr4e6tbv888sz91jc/tinymce/7/tinymce.min.js
```
[the remark this answers is not in the document]
Comment (MitchC1999) [reply antora.yml:13]: I've added a checklist item in the main section of this PR, so I wouldn't forget 😂

**Review by kemister85 — APPROVED** [review]
Nice work, thank you for putting this together.

**Pull request merged**